poetry.lock

```diff
@@ -0,0 +1,458 @@
+[[package]]
```

Comment:
I don't think libraries (as opposed to apps) should check in their lock file, since they are supposed to work with a range of dependency versions

Reply:
Yes, it is not necessary
https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control

Reply:
this is a moot point.
checking or omitting the `poetry.lock` file into/from the repo neither pins the deps where it counts (for library users), nor tests the library across versions of deps (separate CI workflow with a matrix would be needed for that).

Reply:
I'll try to add matrix testing to the other PR

Reply:
Added matrix to test on all supported python versions #6